src/modules/aws_ecr/index.ts

```diff
@@ -200,14 +190,24 @@ class RepositoryImageMapper extends MapperBase<RepositoryImage> {
                   r.repositoryName,
                   r.registryId,
                 );
-                if (ri?.imageIds) {
-                  const imageDetails = await this.getRepositoryImage(
-                    regionClient.ecrClient,
-                    ri.imageIds,
-                    r.repositoryName,
-                  );
-                  if (imageDetails && imageDetails.images) {
-                    for (const image of imageDetails.images) images.push({ ...image, region });
+                if (ri?.imageDetails) {
+                  for (const imageDetail of ri.imageDetails) {
+                    if (imageDetail.imageDigest && imageDetail.repositoryName) {
+                      if (imageDetail.imageTags && imageDetail.imageTags.length > 0) {
+                        for (const imageTag of imageDetail.imageTags) {
+                          out.push(
+                            await this.repositoryImageMapper(
+                              { ...imageDetail, imageTags: [imageTag] },
+                              ctx,
+                              'private',
+                              region,
+                            ),
+                          );
+                        }
+                      } else {
+                        out.push(await this.repositoryImageMapper(imageDetail, ctx, 'private', region));
+                      }
+                    }
```

Comment:
It's a bit hard to follow this nested code. Maybe we can wrap it in a function and early-return instead of `if-else` conditions?